op-supervisor/supervisor/backend/cross/cycle_test.go

```diff
@@ -474,7 +470,10 @@ func TestHazardCycleChecksCycle(t *testing.T) {
 			msg:       "expected cycle detection error for cycle through executing messages",
 		},
 		{
-			name: "3-cycle in single chain",
+			// 1->2->1
+			// 2->3->2
+			// 1->3->2->1
```

Comment:
```suggestion
			// 1->2->1
			// 2->3->2
```
I think you are flipping the initiating/executing side in some of these comments. If the format is `exec -> init` message, then there is no such thing as `1` depending on `3` here. In the test-cases it's `3` that depends on `1`.

Reply:
I get this confusion, so we are saying that 3->1 does not count as cycle. But initially I still included this as a cycle, by merging 2,2 cycles. So 3->1->2->3 for completeness.

(3->1) + (1->2) + (2->3)

Reply:
https://github.com/ethereum-optimism/optimism/pull/15568/files#r2060422854 this comment may be helpful although it uses the reverse notation.

Reply:
Oh I see now, there's indeed that larger cycle, where `(3->1)` is valid, but because `(1->2)` is an invalid pointer into the future, and `(2->3)` is an invalid pointer into the future, we end up with a cycle, even though `(1->2)` is already a cycle because of the `(2->1)` implied by regular order, and likewise for `(2->3)` being a cycle with `(3->2)` on its own.

Reply:
Updated the suggestion with clarification. There are 4 distinct cycles.